driver/src/contracts/stablex_auction_element.rs

```diff
@@ -45,14 +46,12 @@ impl StableXAuctionElement {
         let denominator = BigEndian::read_u128(&bytes[80..96]);
         let remaining = BigEndian::read_u128(&bytes[96..112]);
         let (buy_amount, sell_amount) = compute_buy_sell_amounts(numerator, denominator, remaining);
-        let order_counter = order_count.entry(account_id).or_insert(0);
-        *order_counter += 1;
         StableXAuctionElement {
             valid_from,
             valid_until,
             sell_token_balance,
             order: Order {
-                id: *order_counter - 1,
+                id: 0,
```

Comment:
This seams wrong to me. Should be `*order_counter - 1`.

Reply:
I have considered this but I would like the deserialize function to just deserialize what it can from the input bytes.